include/BufferPool.h

```diff
@@ -23,25 +23,16 @@
  *
  */
 
-#ifndef BUFFER_MANAGER_H
-#define BUFFER_MANAGER_H
+#pragma once
```

Comment:
I saw you're using `#pragma once` instead of macro guards.
I know this has some advantages and most major compilers support this. However, Do we need to change existing files? Also, do we need to consider some compilers which don't support this directive?

Reply:
No need to change existing files. Personally I don't know of any relevant compilers that don't support this and have never had `#pragma once` become a problem on any platform.